error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/MonoZipOfMonoVoidUsageTest.java

```diff
@@ -0,0 +1,118 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class MonoZipOfMonoVoidUsageTest {
+  /**
+   * Line 21 is needed to simulate the unwanted intrinsic operations, which are not intended to be
+   * processed by the rule but will be scanned anyway.
+   */
+  @Test
+  void identification() {
+    CompilationTestHelper.newInstance(MonoZipOfMonoVoidUsage.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "import reactor.core.publisher.Mono;",
+            "",
+            "public final class A<T> {",
```

Comment:
We use generics because the implementation used to complain about some valid use cases, e.g. `ReactorRules`, line 191:
```
@BeforeTemplate
    Mono<Tuple2<T, S>> before(Mono<T> mono, Mono<S> other) {
      return mono.zipWith(other);
    }
```

That's why I intentionally introduced generic test